### ISSUE TITLE
Selecting run with a time range

### ISSUE THREAD

**Issue by bkhelifi** [gammapy/gammapy#2613]
**Gammapy version**
The dev version

**Bug description**
Using the HESS data release, the selection of observations using the table selection function via the criteria 'time_box' crashes. This is linked to the management of the absolute times or relative ones (as defined in the gadf). Both should be accepted somehow.

**Expected behavior**
Whatever the stored time format (relative or absolute), our time selection should work. Two ways are proposed, either via the data_store (using by the LC) or via the obs_table (for any purpose). I propose to keep for the moment these ways (knowing also that their behaviours are different), taking care that both are working

**To Reproduce**
data_store = DataStore.from_dir("$GAMMAPY_DATA/hess-dl3-dr1")
time_selection = dict(type='time_box', time_range=Time(['2006-07-29T00:00:01', '2006-07-30T13:00:00']))
selected_obs_table = data_store.obs_table.select_observations(time_selection)

**Comment by cdeil** [comment]
Complete reproducer:
```
from astropy.time import Time
from gammapy.data import DataStore

data_store = DataStore.from_dir("$GAMMAPY_DATA/hess-dl3-dr1")
time_selection = dict(type='time_box', time_range=Time(['2006-07-29T00:00:01', '2006-07-30T13:00:00']))
selected_obs_table = data_store.obs_table.select_observations(time_selection)
```


Output:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/work/code/gammapy/gammapy/data/obs_table.py", line 274, in select_observations
    "TSTART", selection["time_range"], selection["inverted"]
  File "/Users/deil/work/code/gammapy/gammapy/data/obs_table.py", line 163, in select_time_range
    if self.meta["TIME_FORMAT"] == "absolute":
KeyError: 'TIME_FORMAT'
```

Somewhat related: #1172

Since many years the spec says that the time ref header keys are required in obs table:
https://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/obs_index/index.html#mandatory-header-keywords

It doesn't mention a `TIME_FORMAT` key, so `if self.meta["TIME_FORMAT"] == "absolute"` is no good.

https://github.com/gammapy/gammapy/blob/360f5237a11c6e73ad5dcbf04111cbecd83d68ee/gammapy/data/obs_table.py#L163

I suggest to remove the support for float times (MET) here and everywhere in Gammapy interfaces, just support Time objects or things that convert to time when calling `time = Time(time)` on a `time` input. Supporting float METs randomly in one place is confusing to users, and supporting it consistently in all methods (especially in gammapy.data and gammapy.time) would be a lot of work?

@registerrier - Thoughts? Can you send a PR or should I take a stab at it?

**Comment by registerrier** [comment]
I can correct the current issue, no problem.
Regarding the support for METs, it seems they are always used to store data (the obs table, the GTI and the light curve all use a reference and an elapse time). This should be transparent to users indeed and all functions and methods should use `Time`, which I think is the case most of the time.

Part of the issue is that the object in memory is the same as the one stored on disk and defined in the specs such as gadf. It could be different, but changing this is quite a bit of work.

**Comment by cdeil** [comment]
@registerrier - Indeed, there is the big question whether to store `Time` objects in tables or not. This has been discussed extensively and several times in the past years. We can again, if you prefer to store `Time` objects and go that way, and volunteer to implement it now.

Otherwise, my suggestion here would be to change `ObservationTable` to work like `Eventlist`:
https://docs.gammapy.org/0.14/_modules/gammapy/data/event_list.html#EventListBase.select_time
Add a property that maps the table column(s) to `Time` object, and then copy & paste the implementation from `EventList.select_time`.

I'm sure other `ObservationTable` methods have similar issues, e.g. for region selection, see #1172
Basically `ObservationTable` needs to be overhauled and ~ 1 day of work, its implementation should match `EventList`, avoiding the code duplication isn't easy. Of course, OK to improve incrementally and just fix the time selection for now.

**Comment by registerrier** [comment]
This specific issue is solved by PR #2618. 

Keeping issue open to remember that testing is not done is a very solid manner in the `ObservationTable`: e.g. random table are created and the selection formula are tested against themselves. The test should use simple values with explicit solutions instead.

The choice of time interval selection approach (with start or stop or both fully inside time range) should be clarified and made more uniform inside gammapy.
 
This should be cleaned-up and refactored for v1.0

**Comment by bkhelifi** [comment]
Hi @registerrier ,
I have not yet understood which kind of issue you are thinking in order to improve the tests. 
The function ``test_select_time_box`` seems to test simple values after a time selection....
This issue seems ready to be closed, is not it?

**Comment by registerrier** [comment]
The test there is a bit complex and uses an assert that follows closely the implementation it should test. 
This could be improved but this is likely not a reason to keep the issue open for so long.  Technically the issue with testing is there for most tests in `ObservationTable`.

Closing.
